clients/stellar-relay-lib/src/connection/connector/message_reader.rs

```diff
@@ -93,18 +79,8 @@ async fn read_message_from_stellar(stream_clone: Arc<Mutex<TcpStream>>) -> Resul
 		// check whether or not we should read the bytes as:
 		// 1. the length of the next stellar message
 		// 2. the remaining bytes of the previous stellar message
-		// Temporary scope for locking
-		let result = {
-			let mut stream = stream_clone.lock().unwrap();
-			stream.read(&mut buff_for_reading)
-		};
-
-		match result {
-			Ok(size) if size == 0 => {
-				// No data available to read
-				tokio::task::yield_now().await;
-				continue
-			},
+		match connector.tcp_stream.read(&mut buff_for_reading).await {
+			Ok(size) if size == 0 => continue,
```

Comment:
We ended up not needing `yield_now` here? that is great news! That must mean that `async_std` is indeed compatible with Tokio.

Reply:
yes!!! it actually was because of https://github.com/pendulum-chain/spacewalk/pull/481#discussion_r1467226284.